web/pgadmin/authenticate/oauth2.py

```diff
@@ -159,12 +159,16 @@ def login(self, form):
                 self.oauth2_current_client
             ]['OAUTH2_ADDITIONAL_CLAIMS']
 
-        (valid, reason) = self.__is_additional_claims_valid(profile,
-                                                            additinal_claims)
-
-        if not valid:
+        (valid_profile, reason) = self.__is_any_claims_valid(profile,
+                                                             additinal_claims)
+        
+        id_token_claims = session['oauth2_token']['userinfo']
```

Comment:
somehow `profile` variable is the userinfo endpoint response, that is not to be confused with `session['oauth2_token']['userinfo']`. the `session['oauth2_token']['userinfo']` contains the id_token parsed information already validated with the jwks public key. 

https://github.com/lepture/authlib/blob/master/authlib/integrations/flask_client/apps.py#L104